qemu/tests/virtio_serial_various_chardev_hotplug.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from virttest import error_context
+from qemu.tests.virtio_console import add_chardev
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Hot-plug Various chardev
+
+    1. Boot a guest without chardev, no serial port & no pci
+    2. Hot plug Unix chardev backend
+    3. Hot plug udp chardev backend
+    4. Hot plug null backend
+    5. Hot plug file backend
+    6. Hot plug pty backend
+    7. Hot plug ringbuf backend
+    8. Write data to ringbuf
+    9. Read data from ringbuf
+    10. Hot-unplug Unix chardev backend
+    11. Hot-unplug udp chardev backend
+    12. Hot-unplug null backend
+    13. Hot-unplug file backend
+    14. Hot-unplug pty backend
+    15. Hot-unplug ringbuf backend
+
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def cmd_qmp_log(vm, cmd, args):
+        logging.debug("[qmp cmd %s] %s" % (cmd, args))
+        reply = vm.monitor.cmd_qmp(cmd, args)
+        logging.debug("[qmp reply] %s" % reply)
+        if "error" in reply:
+            if reply["error"]["class"] == "CommandNotFound":
```

Comment:
Line 37 seems conflict with Line 38, do you think `reply` is a string or dictionary?

Reply:
reply is a Dictionary, Line 38 wants to check qmp command error but not command format error.

Reply:
> Line 37 seems conflict with Line 38, do you think `reply` is a string or dictionary?

reply is a Dictionary, Line 38 wants to check qmp command error but not command format error.